### PR TITLE
vdk-jupyter: add create job command to jupyter

### PR DESCRIPTION
**Pull request by duyguHsnHsn**
What:
Added front-end components of Create Job option to the VDK menu in JupyterLab, requests to the server extension for creating a job and handling the operation in the server. Currently, it creates a standard data job with .sql and .py files.

Why:
It is linked to https://github.com/vmware/versatile-data-kit/issues/1282 and https://github.com/vmware/versatile-data-kit/issues/1320

Testing: Manual tests by reinstalling the extension

Signed-off-by: Duygu Hasan [hduygu@vmware.com](mailto:hduygu@vmware.com)

### CODE REVIEW

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py [hunk 1]
```diff
@@ -68,3 +70,28 @@ def download_job(name: str, team: str, rest_api_url: str, path: str):
...
         return f"Downloaded the job with name {name} to {path}. "
+
+    # TODO: make it work with notebook jobs
+    @staticmethod
+    def create_job(
+        name: str, team: str, rest_api_url: str, path: str, local: bool, cloud: bool
+    ):
+        """
+        Execute `download job`.
+        :param name: the name of the data job that will be created
```
Comment (ivakoleva) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py:81]: Is the expectation like `vdk create` or `vdk download-job`?

**Review by ivakoleva — APPROVED** [review]
LGTM overall;
maybe we need to think early on about automated testing

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py [hunk 2]
```diff
@@ -68,3 +70,28 @@ def download_job(name: str, team: str, rest_api_url: str, path: str):
         cmd = JobDownloadSource(rest_api_url)
         cmd.download(team, name, path)
         return f"Downloaded the job with name {name} to {path}. "
+
+    # TODO: make it work with notebook jobs
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py:74]: Current implementation creates a standard data job with .py and .sql files and the user should delete them and create a notebook instead. We can add a sample job which consists of .ipynb file instead.

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts [hunk 3]
```diff
@@ -138,3 +138,47 @@ export async function downloadJobRequest() {
...
+ * The information about the data job is retrieved from sessionStorage and sent as JSON.
+ */
+ export async function createJobRequest() {
+  let dataToSend = {
+    jobName: sessionStorage.getItem('create-job-name'),
+    jobTeam: sessionStorage.getItem('create-job-team'),
+    restApiUrl: sessionStorage.getItem('create-job-rest-api-url'),
+    jobPath: sessionStorage.getItem('create-job-path'),
+    isLocal: sessionStorage.getItem('local'),
+    isCloud: sessionStorage.getItem('cloud')
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:153]: I am not sure whether I understand the first question properly. For the memory state model, I have created an issue, going to work on it pretty soon

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py (hunk 1 above)
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py:81]: Oh, sorry I copy pasted the comment from the download job method and edited it but I missed this one. I will change it to "create job"

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (murphp15) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:153]: yeah it my opinion there should only be one variable. isCloud. if it is false then that means that it is local.

**Review by murphp15 — APPROVED** [review]

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:153]: hm, but what if it is both?

[the remark this answers is not in the document]
Comment (murphp15) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:153]: oh they want to save it locally and in the cloud?

[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:153]: It is an option in the CLI, so I will leave it like that

**Pull request merged**